src/codegen/utilities/function_helpers.py

```diff
@@ -0,0 +1,207 @@
+"""This contains the helper methods used in functions generation."""
+from codegen.functions.function import Function
+from codegen.utilities.helpers import camel_to_snake_case
+
+FUNCTION_NAME_CHANGE_SET = {
+    "TEDSAIRTD": "TEDS_AI_RTD",
+    "TEDSAI": "TEDS_AI",
+    "AIRTD": "AI_RTD",
+    "CIGPS": "CI_GPS",
+}
+
+
+def get_functions(metadata, class_name):
+    """Converts the scrapigen metadata into a list of functions."""
+    all_functions = metadata["functions"]
+    functions_metadata = []
+    for function_name, function_data in all_functions.items():
+        if (
+            "python_class_name" in all_functions[function_name]
+            and all_functions[function_name]["python_class_name"] == class_name
+        ):
+            function_data["c_function_name"] = function_name
```

Comment:
"get" functions usually shouldn't modify their inputs. Please copy the dictionary before modifying it.

Reply:
Applied `deepcopy()` for both functions & attributes metadata.

Reply:
`get_attributes` doesn't need to copy because it doesn't modify the metadata.

Reply:
Resolved.